beancount_import/training.py

```diff
@@ -104,11 +106,15 @@ def __init__(
             self,
             account_source_map: Dict[str, 'Source'],
             sources: Iterable['Source'],
+            skip_accounts: Set[str],
             ignore_account_pattern:
             str = DEFAULT_IGNORE_ACCOUNT_FOR_CLASSIFICATION_PATTERN,
     ) -> None:
         self.account_source_map = account_source_map
-        self.ignore_account_pattern = ignore_account_pattern
+        self.skip_accounts_re = [re.compile(p) for p in
```

Comment:
You can combine it into a single regular expression (using `(?: )` and `|`), which should be much more efficient to evaluate.

Reply:
Okay, that's better and simpler. Done.